lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4SharedOnChipRam.scala

```diff
@@ -60,60 +61,137 @@ case class Axi4SharedOnChipRam(dataWidth : Int, byteCount : BigInt, idWidth : In
 }
 
 
-//WARNING, do not support backpresure
-case class Axi4SharedOnChipRamMultiPort(portCount : Int, dataWidth : Int,byteCount : BigInt,idWidth : Int) extends Component{
-  val axiConfig = Axi4SharedOnChipRam.getAxiConfig(dataWidth,byteCount,idWidth)
-
-  val io = new Bundle {
-    val axis = Vec(slave(Axi4Shared(axiConfig)), portCount)
-  }
+object Axi4SharedOnChipRamPort {
+    def apply(config: Axi4Config, ram: Mem[Bits]) = {
+        val port = new Axi4SharedOnChipRamPort(config)
+        port.attach(ram)
+        port
+    }
+
+    def apply(config: Axi4Config, ram: Mem[Bits], softReset: Bool) = {
+        val clock = ClockDomain.current.copy(softReset = softReset)
+        val domain = new ClockingArea(clock) {
+            val port = new Axi4SharedOnChipRamPort(config)
+            port.attach(ram)
+        }
+        domain.port
+    }
+}
 
-  val wordCount = byteCount / axiConfig.bytePerWord
-  val ram = Mem(axiConfig.dataType,wordCount.toInt)
-  val wordRange = log2Up(wordCount) + log2Up(axiConfig.bytePerWord)-1 downto log2Up(axiConfig.bytePerWord)
+case class Axi4SharedOnChipRamPort(config: Axi4Config) extends ImplicitArea[Axi4Shared] {
+    val axi                                = Axi4Shared(config)
+    override def implicitValue: Axi4Shared = this.axi
+
+    val ram = Handle[Mem[Bits]]
+
+    val logic = Handle {
+        new Area {
+            val wordRange       = config.wordRange
+            val arw             = axi.arw.unburstify
+            val writeDataStream = axi.writeData
+
+            val readStream      = Stream(Axi4R(axi.config))
+            val readAddrStream  = cloneOf(arw)
+            val writeStream     = Stream(Axi4B(axi.config))
+            val writeAddrStream = cloneOf(arw)
+
+            Vec(readAddrStream, writeAddrStream) <> StreamDemux(
```

Comment:
Why forking the stream on the address side ? instead of forking it at the response side ? 
Was the bug of the original implementation related to the unburstify ?